packages/ui/src/components/Tabs/STab.vue

```diff
@@ -0,0 +1,138 @@
+<script lang="ts">
+export default defineComponent({
+  name: 'STab',
+})
+</script>
+
+<script setup lang="ts">
+import { useTabsStateApi } from './api'
+
+const props = withDefaults(
+  defineProps<{
+    disabled?: boolean
+    borderRadius?: string
+  }>(),
+  {
+    disabled: false,
+    borderRadius: '',
+  },
+)
+
+const instance = getCurrentInstance()
+const { active, tabs, selectTab } = useTabsStateApi() || {
+  active: ref(0),
+  tabs: ref<any>([]),
+  selectTab: (t: number) => {},
+}
+
+const index = computed(() => tabs.value.findIndex((target: any) => target.uid === instance?.uid))
+const isActive = computed(() => index.value === active.value)
+
+const activateTab = () => {
+  selectTab(index.value)
+}
+
+watchEffect(() => {
+  if (index.value === -1) {
+    if (instance) tabs.value.push(instance)
+  }
+})
+</script>
+
+<template>
+  <button
+    role="tab"
+    class="s-tab flex justify-center items-center sora-tpg-p2"
+    :disabled="disabled"
+    :class="[
+      { 's-tab_active': isActive },
+      borderRadius ? `s-tab_border-radius-${borderRadius}` : 's-tab_border-radius-none',
+    ]"
+    @click="activateTab"
+  >
+    <div class="s-tab__label-container flex justify-center items-center">
+      <div class="s-tab__label">
+        <slot />
+      </div>
+    </div>
+  </button>
+</template>
+
+<style lang="scss">
+@use '@/theme';
+$border-radius-side: v-bind(borderRadius);
+
+$inactive-font-color: theme.token-as-var('sys.color.content-primary');
+$background-color: theme.token-as-var('sys.color.background');
+$hover-font-color: theme.token-as-var('sys.color.primary-hover');
+$text-color-active: theme.token-as-var('sys.color.content-primary');
+$text-color-disabled: theme.token-as-var('sys.color.content-quaternary');
+$active-tab-shadow: theme.token-as-var('sys.shadow.active-tab');
+
+.s-tab {
+  @apply select-none;
+  padding: 4px;
+
+  background: $background-color;
+  color: $inactive-font-color;
+  outline: none;
+
+  &_border-radius-none {
+    border-radius: 0;
+  }
+
+  &_border-radius-right {
+    border-radius: 0 8px 8px 0;
+  }
+
+  &_border-radius-left {
+    border-radius: 8px 0 0 8px;
+  }
+
+  &_border-radius-full {
+    border-radius: 8px;
+  }

```

Comment:
Naming for the `border-radius` modifier does not satisfy BEM spec: `<element>_<modifier>_<modifier value>`. So, it would be correct to use `s-tab_border-radius_full` etc, not `s-tab_border-radius-full`.